### ISSUE TITLE
Implement struct StringInstrumentModel

### ISSUE THREAD

**Issue by jsbean** [jsbean/DNM#202]
```Swift
struct StringInstrumentModel {
    let stringModels: [StringModel]
    init(stringsWithFundamentalPitches: [Pitch]) { ... }
    init(stringsWithFundamentals: [Float]) { ... }
}

struct StringModel {
    let fundamentalPitch: Pitch
     
    // TODO: implement isPrime
    func nodesForPartial(partial: Int) -> [StringNode] { 
        guard partial >= 2 else { return [] }
        // filters out redundant partial instances (not primes)
        return Array(1...(n-1)).filter { $0.isPrime }.map { StringNode(f, partial, $0 }
    }

    func nodesForPartialsInRange(_ min: , _ max:) -> [StringNode] { get ... }
}

// use also for artificial harmonic fingerings
struct StringNode {
    let fundamentalPitch: Pitch
    let partialNumber: Int
    let partialInstance: Int
    var soundingPitch: Pitch { get ... }
    var fingeredPitch: Pitch { get ... }
    init(_,_,_) { }
}
```

**Comment by jsbean** [comment]
For example, find fingered pitch location for 7th harmonic on open string:

```Swift
let string = StringModel(...)
string.nodesForPartial(7).first!.fingeredPitch
```

**Comment by jsbean** [comment]
Closed with 3b89152c56ba51fc56eab574ea17ac0accbc5680
